core/src/test/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotServiceSpec.scala

```diff
@@ -108,6 +127,92 @@ class SnapshotServiceSpec extends AnyWordSpecLike with Matchers with MockitoSuga
+
+      val mockWorkspaceManagerDAO = defaultMockWorkspaceManagerDao()
+      when(
+        mockWorkspaceManagerDAO.getWorkspace(any[UUID], any[RawlsRequestContext])
+      ).thenAnswer(_ => new ApiException(404, "Workspace does not exist"))
+
+      val mockDataRepoDAO: DataRepoDAO = new MockDataRepoDAO("mockDataRepo")
+
+      val workspace = minimalTestData.workspace
+
+      val snapshotService = SnapshotService.constructor(
+        slickDataSource,
+        mockSamDAO,
+        mockWorkspaceManagerDAO,
+        "fake-terra-data-repo-dev",
+        mockDataRepoDAO
+      )(testContext)
+
+      // call createSnapshot on the service
+      Await.result(
+        snapshotService.createSnapshot(workspace.toWorkspaceName,
+                                       NamedDataRepoSnapshot(DataReferenceName("foo"),
+                                                             Option(DataReferenceDescriptionField("foo")),
+                                                             UUID.randomUUID()
+                                       )
+        ),
+        Duration.Inf
+      )
+
+      // assert that the service checked to see if the workspace exists
+      verify(mockWorkspaceManagerDAO, times(1)).getWorkspace(
+        any[UUID],
+        any[RawlsRequestContext]
+      )
+
+      // assert that the service called WSM's createWorkspace
+      verify(mockWorkspaceManagerDAO, times(1)).createWorkspace(
+        any[UUID],
+        any[RawlsRequestContext]
+      )
+    }
+
+    "not create a WSM workspace if one already exists when creating a snapshot reference" in withMinimalTestDatabase {
+      _ =>
+        val mockSamDAO = defaultMockSamDao()
+
+        val mockWorkspaceManagerDAO = defaultMockWorkspaceManagerDao()
+
+        val mockDataRepoDAO: DataRepoDAO = new MockDataRepoDAO("mockDataRepo")
+
+        val workspace = minimalTestData.workspace
+
+        val snapshotService = SnapshotService.constructor(
+          slickDataSource,
+          mockSamDAO,
+          mockWorkspaceManagerDAO,
+          "fake-terra-data-repo-dev",
+          mockDataRepoDAO
+        )(testContext)
+
+        // call createSnapshot on the service
+        Await.result(
+          snapshotService.createSnapshot(workspace.toWorkspaceName,
+                                         NamedDataRepoSnapshot(DataReferenceName("foo"),
+                                                               Option(DataReferenceDescriptionField("foo")),
+                                                               UUID.randomUUID()
+                                         )
+          ),
+          Duration.Inf
+        )
+
+        // assert that the service checked to see if the workspace exists
+        verify(mockWorkspaceManagerDAO, times(1)).getWorkspace(
+          any[UUID],
+          any[RawlsRequestContext]
+        )
+
+        // assert that the service DID NOT call WSM's createWorkspace
+        verify(mockWorkspaceManagerDAO, times(0)).createWorkspace(
```

Comment:
In the opposite direction from my other comment...this is appropriate use of the `any` because we want to make sure it didn't get called _at all_ or _with anything_